src/engine/routing_algorithms/many_to_many_ch.cpp

```diff
@@ -240,74 +240,11 @@ void calculateDistances(typename SearchEngineData<ch::Algorithm>::ManyToManyQuer
         }
         if (!packed_leg.empty())
         {
-            auto annotation =
+            EdgeDistance annotation =
                 ch::calculateEBGNodeAnnotations(facade, packed_leg.begin(), packed_leg.end());
+            adjustPathDistanceToPhantomNodes(packed_leg, source_phantom, target_phantom, annotation);
```

Comment:
Here: not sure if out-ref is a good idea. Maybe pass in and take return val again?